src/ddsrt/include/dds/ddsrt/ifaddrs.h

```diff
@@ -36,15 +43,38 @@ struct ddsrt_ifaddrs {
 
 typedef struct ddsrt_ifaddrs ddsrt_ifaddrs_t;
 
+/**
+ * @brief Get the interface addresses (@ref ddsrt_ifaddrs) for specific address families
+ * 
+ * Note that array 'afs' must be terminated with an element 'DDSRT_AF_TERM'.
+ * 
+ * @param[out] ifap interface address pointer
+ * @param[in] afs an array of address families
+ * @return a DDS_RETCODE
+ */
 dds_return_t
 ddsrt_getifaddrs(
   ddsrt_ifaddrs_t **ifap,
   const int *afs);
 
+/**
+ * @brief Free the interface addresses
+ * 
+ * @param[in] ifa the interface addresses to free
+ */
 void
 ddsrt_freeifaddrs(
   ddsrt_ifaddrs_t *ifa);
 
+/**
+ * @brief Get the mac address for a given interface name
+ * 
+ * Copies the first 6 bytes of the mac address into the buffer 'mac_addr'.
+ * 
+ * @param[in] interface_name the name of the interface
+ * @param[out] mac_addr a buffer to copy the mac address into
+ * @return a DDS_RETCODE
```

Comment:
It would be good to note that this is one of `OK`, `NOT_ALLOWED`, `OUT_OF_RESOURCES` or simply `ERROR` and that specifying an unknown interface simply results in `ERROR`.

If the interface exists and doesn't actually have a MAC address, like a loopback interface, the behaviour is platform-dependent. (On Linux, you get all 0, but that may not be universally true.)